### DISCUSSION TITLE
Set media after page rendered

### DISCUSSION THREAD

**Discussion by SolarGuy8** [katspaugh/wavesurfer.js#3564] [Q&A]
Hello, Dear Community and Author!

I really need help ASAP.

Wavesurfer is one of the best libraries, but I've encountered an issue and don't know how to fix it.

I have a shared player in my app. This player is used in many places to sync track playback progress.

My current code:

```
const wavesurferObject = WaveSurfer.create({
  container: containerRef.current,
  waveColor: gradient,
  progressColor: progressGradient,
  barWidth: 3,
  url: audioUrl + '?not-from-cache-please',
  dragToSeek: true,
  height: 'auto',
  normalize: true,
  media: player.getElement(),
});
```

However, when, for example, track A is playing, and I open the page for track B, my wavesurfer gets rerendered, and track A stops playing because wavesurfer updates the `media` and replace track (mini-player moved to start position). I've tried to set the `media` only when the current track page starts playing using this code:

```
const [trackInitiated, setTrackInitiated] = useState<boolean>(false);

const { active } = useTrackStatus(uid);

useEffect(() => {
  if (active && !trackInitiated) {
    wavesurfer?.setOptions({
      media: player.getElement(),
    });
    setTrackInitiated(true);
  }
}, [active, trackInitiated]);
```

But it doesn't take any effect. The waveform just doesn't play. If I set `wavesurfer.setMediaElement(player.getElement())`, it just hangs and starts consuming a lot of memory.

How can I make it so that the waveform is displayed, but the media element is not changed until I press play on the current track?

**Comment by katspaugh** [comment]
Make sure WaveSurfer.create is only run once. You might have more luck with the `@wavesurfer/react` package which is optimized to not re-render needlessly: https://wavesurfer.xyz/examples/?react.js

**Reply by SolarGuy8** [reply]
Yeah, but it's doesn't support typescript 4.0.2 :( 

What I should to do in this situation?
```
  const wavesurfer = useWaveSurfer(
    waveformRef,
    audioUrl,
    gradient,
    progressGradient,
  );

  const [trackInitiated, setTrackInitiated] = useState<boolean>(false);

  const { active } = useTrackStatus(uid);

  useEffect(() => {
    if (active && !trackInitiated) {
      wavesurfer.empty();
      wavesurfer.setMediaElement(player.getElement());
      wavesurfer.play();
      setTrackInitiated(true);
    }
  }, [active, trackInitiated]);
```

and here I have a memory leaks

**Reply by SolarGuy8** [reply]
Wavesurfer calls only once per page rendering. Memory leaks start when I `setMediaElement()`

**Reply by SolarGuy8** [reply]
But is it any way, how can I update MediaElement after wavesurfer creation?
